src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -0,0 +1,143 @@
+import { createHash } from "crypto";
+
+import { K8s, Log, kind } from "pepr";
+import { UDSConfig } from "../../../../config";
+import { Client } from "../types";
+import { buildChain } from "./authservice";
+import { Action, AuthserviceConfig } from "./types";
+
+export const operatorConfig = {
+  namespace: "authservice",
+  secretName: "authservice-uds",
+  baseDomain: `https://sso.${UDSConfig.domain}`,
+  realm: "uds",
+};
+
+export async function setupAuthserviceSecret() {
+  if (process.env.PEPR_WATCH_MODE === "true" || process.env.PEPR_MODE === "dev") {
+    Log.info("One-time authservice secret initialization");
+    // create namespace if it doesn't exist
+    await K8s(kind.Namespace).Apply({
+      metadata: {
+        name: operatorConfig.namespace,
+      },
+    });
+
+    // create secret if it doesn't exist
+    try {
+      const secret = await K8s(kind.Secret)
+        .InNamespace(operatorConfig.namespace)
+        .Get(operatorConfig.secretName);
+      Log.info(`Authservice Secret exists, skipping creation - ${secret.metadata?.name}`);
+    } catch (e) {
+      Log.info("Secret does not exist, creating authservice secret");
+      try {
+        await updateAuthServiceSecret(buildInitialSecret(), false);
+      } catch (err) {
+        Log.error(err, "Failed to create UDS managed authservice secret.");
+      }
```

Comment:
The added log message is good, but this should still cause the entire reconcile to fail.

Maybe re-throw or wrap the exception and throw it. I'm not sure how you all are handling exceptions.

Reply:
Re-throwing this now as it is a critical failure.